File: Python/client_template.py
import discord
from discord.ext import commands, tasks
import datetime, asyncio

TOKEN = "TOKENCLIENT" # Ne pas toucher ceci et remplacer automatiquement par le main.py
EXPIRY_DATE = "2099-12-31 23:59:59" # Date d'expiration du bot.'

intents = discord.Intents.default()
intents.message_content = True
intents.members = True
bot = commands.Bot(command_prefix="?", intents=intents)

status_cycle = ["En ligne", "Prêt à aider", "Prefix ?"]

def expired():
    return datetime.datetime.utcnow() > datetime.datetime.strptime(EXPIRY_DATE, "%Y-%m-%d %H:%M:%S")

@bot.event
async def on_ready():
    print(f"{bot.user} prêt | Expiration : {EXPIRY_DATE}")
    change_status.start()
    check_expiry.start()

@tasks.loop(seconds=15)
async def change_status():
    current = status_cycle.pop(0)
    await bot.change_presence(activity=discord.Game(current))
    status_cycle.append(current)

@tasks.loop(minutes=1)
async def check_expiry():
    if expired():
        print(f"{bot.user} expiré, déconnexion...")
        await bot.close()

@bot.command()
async def ping(ctx):
    await ctx.send(f"Pong! {round(bot.latency*1000)}ms")

@bot.command()
async def support(ctx):
    await ctx.send("Pour toute aide, contacte le serveur support.")

@bot.command()
async def help(ctx):
    embed = discord.Embed(title="Commandes disponibles", color=0x3498db)
    embed.add_field(name="?ping", value="Test du bot", inline=False)
    embed.add_field(name="?support", value="Lien du support", inline=False)
    embed.add_field(name="?ban", value="Bannir un membre", inline=False)
    embed.add_field(name="?kick", value="Expulser un membre", inline=False)
    await ctx.send(embed=embed)

@bot.command()
async def ban(ctx, member: discord.Member, *, reason=None):
    try:
        await member.ban(reason=reason)
        await ctx.send(f"{member} a été banni.")
    except:
        await ctx.send("Impossible de bannir ce membre.")

@bot.command()
async def kick(ctx, member: discord.Member, *, reason=None):
    try:
        await member.kick(reason=reason)
        await ctx.send(f"{member} a été expulsé.")
    except:
        await ctx.send("Impossible d'expulser ce membre.")

bot.run(TOKEN)

File: JavaScript/client_template.js
const { Client, GatewayIntentBits, Partials, Collection, EmbedBuilder } = require('discord.js');
const TOKEN = "TOKENCLIENT"; // Ne pas toucher !!
const EXPIRY_DATE = "2099-12-31 23:59:59"; // Date d'expiration du bot

const client = new Client({
    intents: [
        GatewayIntentBits.Guilds,
        GatewayIntentBits.GuildMessages,
        GatewayIntentBits.MessageContent,
        GatewayIntentBits.GuildMembers
    ],
    partials: [Partials.Channel]
});

const statusCycle = ["En ligne", "Prêt à aider", "Prefix ?"];
let statusIndex = 0;

function isExpired() {
    return new Date() > new Date(EXPIRY_DATE);
}

client.once('ready', () => {
    console.log(`${client.user.tag} prêt | Expiration : ${EXPIRY_DATE}`);
    setInterval(changeStatus, 15000);
    setInterval(checkExpiry, 60000);
});

function changeStatus() {
    client.user.setActivity(statusCycle[statusIndex], { type: 0 });
    statusIndex = (statusIndex + 1) % statusCycle.length;
}

function checkExpiry() {
    if (isExpired()) {
        console.log(`${client.user.tag} expiré, déconnexion...`);
        client.destroy();
    }
}

client.on('messageCreate', async message => {
    if (!message.guild || message.author.bot) return;
    const args = message.content.trim().split(/ +/g);
    const cmd = args.shift().toLowerCase();

    if (cmd === "?ping") {
        message.channel.send(`Pong! ${client.ws.ping}ms`);
    }

    if (cmd === "?support") {
        message.channel.send("Pour toute aide, contacte le serveur support.");
    }

    if (cmd === "?help") {
        const embed = new EmbedBuilder()
            .setTitle("Commandes disponibles")
            .addFields(
                { name: "?ping", value: "Test du bot", inline: false },
                { name: "?support", value: "Lien du support", inline: false },
                { name: "?ban", value: "Bannir un membre", inline: false },
                { name: "?kick", value: "Expulser un membre", inline: false }
            )
            .setColor(0x3498db);
        message.channel.send({ embeds: [embed] });
    }

    if (cmd === "?ban") {
        const member = message.mentions.members.first();
        if (!member) return message.channel.send("Mentionne un membre à bannir.");
        try {
            await member.ban({ reason: args.join(" ") || "Aucune raison fournie" });
            message.channel.send(`${member.user.tag} a été banni.`);
        } catch {
            message.channel.send("Impossible de bannir ce membre.");
        }
    }

    if (cmd === "?kick") {
        const member = message.mentions.members.first();
        if (!member) return message.channel.send("Mentionne un membre à expulser.");
        try {
            await member.kick(args.join(" ") || "Aucune raison fournie");
            message.channel.send(`${member.user.tag} a été expulsé.`);
        } catch {
            message.channel.send("Impossible d'expulser ce membre.");
        }
    }
});

client.login(TOKEN);